libraries/mbed/targets/hal/TARGET_WIZNET/TARGET_W7500x/i2c_api.c

```diff
@@ -0,0 +1,316 @@
+
+    // wait before init
+    I2C_ConfigStruct conf;
+
+    conf.mode = I2C_Master;
+    conf.master.timeout = LONG_TIMEOUT;
+
+    // Common settings: I2C clock = 48 MHz, Analog filter = ON, Digital filter coefficient = 0
+    switch (hz) {
+        case 100000:
+            conf.master.prescale  = 0x61;      // Standard mode with Rise Time = 400ns and Fall Time = 100ns
+            break;
+        case 400000:
+            break;
+        case 1000000:
+            break;
+        default:
+            break;
+    }
+
+    // I2C configuration
+    I2C_Init(I2cHandle, conf);
+}
+
+inline int i2c_start(i2c_t *obj)
+{
+    obj->is_setAddress = 0;
+
+    return 0;
+}
+
+inline int i2c_stop(i2c_t *obj)
+{
+    I2cHandle = (I2C_TypeDef *)(obj->i2c);
+
+    // Generate the STOP condition
+    I2C_Stop(I2cHandle);
+    I2C_Reset(I2cHandle);
+    obj->is_setAddress = 0;
+
+    return 0;
+}
+
+int i2c_read(i2c_t *obj, int address, char *data, int length, int stop)
+{
+    I2cHandle = (I2C_TypeDef *)(obj->i2c);
+    int count;
+    int value;
+
+    if(!obj->is_setAddress)
+    {
+       if( I2C_Start(I2cHandle, address, I2C_READ_SA7) == ERROR )
+        {
+            return -1;
+        }
+       obj->is_setAddress = 1;
+       obj->ADDRESS = address;
+    }
+    else
+    {
+        I2C_Restart_Structure(I2cHandle, address, I2C_READ_SA7);
+        obj->ADDRESS = address;
+    }
+
+    // Read all bytes
+    for (count = 0; count < (length-1); count++) {
+        if( (value = i2c_byte_read(obj, 0)) == -1) return value;
+        data[count] = (char)value;
+    }
+
+    if(stop){
+        if( (value = i2c_byte_read(obj, 1)) == -1) return value;
+        data[count] = (char)value;
+    }
+    else{
+        if( (value = i2c_byte_read(obj, 0)) == -1) return value;
+        data[count] = (char)value;
+    }
+
+    return count;
+}
+
+int i2c_write(i2c_t *obj, int address, const char *data, int length, int stop)
+{
+    I2cHandle = (I2C_TypeDef *)(obj->i2c);
+    int count;
+
+    if(!obj->is_setAddress)
+    {
+       if( I2C_Start(I2cHandle, address, I2C_WRITE_SA7) == ERROR )
+        {
+            return -1;
+        }
+       obj->is_setAddress = 1;
+       obj->ADDRESS = address;
+    }
+    else
+    {
+        I2C_Restart_Structure(I2cHandle, address, I2C_WRITE_SA7);
+        obj->ADDRESS = address;
+    }
+
+    for (count = 0; count < length; count++) {
+        i2c_byte_write(obj, data[count]);
+    }
+
+    // If not repeated start, send stop
+    if (stop) {
+        i2c_stop(obj);
+    }
+
+    return count;
+}
+
+int i2c_byte_read(i2c_t *obj, int last)
+{
+    I2cHandle = (I2C_TypeDef *)(obj->i2c);
+
+    return I2C_ReceiveData(I2cHandle, last);
+}
+
+int i2c_byte_write(i2c_t *obj, int data)
+{
+    I2cHandle = (I2C_TypeDef *)(obj->i2c);
+    return I2C_SendDataAck(I2cHandle,(uint8_t)data);
+}
+
+void i2c_reset(i2c_t *obj)
+{
+    I2cHandle = (I2C_TypeDef *)(obj->i2c);
+
+    I2C_Reset(I2cHandle);
+}
+
+//#if DEVICE_I2CSLAVE
```

Comment:
DEVICE_I2CSLAVE is enabled in the device header but implementation is commented out?